src/selected_role.rs

```diff
@@ -22,17 +24,21 @@ impl SelectedRole {
     pub fn new(ty: RoleType, role: Option<String>) -> Self {
         SelectedRole { ty, role }
     }
+}
+
+impl FromStr for SelectedRole {
+    type Err = crate::Error;
 
-    pub fn from_str(s: &str) -> Option<Self> {
-        let cap = PATTERN.captures(s)?;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let cap = PATTERN.captures(s).unwrap();
```

Comment:
Maybe explain why the unwrap is safe here?

Reply:
It's not, I'll propagate the error upwards